constructr-akka/src/main/scala/de/heikoseeberger/constructr/akka/Constructr.scala

```diff
@@ -69,9 +70,17 @@ final class Constructr private extends Actor with ActorLogging with ActorSetting
   override def receive = Actor.emptyBehavior
 
   private def active(machine: ActorRef): Receive = {
-    case Terminated(`machine`) =>
-      log.error("Terminating the system, because constructr-machine has terminated!")
-      context.system.terminate()
+    case Terminated(`machine`) => {
+      val cluster = Cluster(context.system)
+      val self = cluster.selfAddress
+      if (cluster.state.members.exists(member => member.address == self && member.status == Up)) {
+        log.error("Leaving the cluster, because constructr-machine has terminated!")
+        cluster.leave(self)
```

Comment:
the cluster.leave will trigger the message MemberRemoved that you are already handling by terminating the actor system. So in the end a cluster.leave will trigger a termination of the actor system.